packages/block-library/src/block/block.json

```diff
@@ -24,6 +24,7 @@
 		"html": false,
 		"inserter": false,
 		"renaming": false,
+		"blockVisibility": false,
```

Comment:
I'm curious about the reasoning for not allowing synced patterns to be hidden. I think that might be quite a handy use case (same for template parts which are essentially another form of pattern).

It's something that can be revised later quite easily so doesn't necessarily need to be addressed in this PR.